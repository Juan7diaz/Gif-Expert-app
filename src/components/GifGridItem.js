import React from 'react'

const GifGridItem = ( {title, url} ) => {

    return (

        <div className="card animate__animated animate__backInLeft" >
            <img src={url } alt = { title } />
            <div className="card-body">
                <p className="card-text">{title}</p>
            </div>
        </div>
    )
}

export default GifGridItem
