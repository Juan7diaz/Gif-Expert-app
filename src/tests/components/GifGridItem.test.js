import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import GifGridItem from '../../components/GifGridItem';


describe('Prueba en el componente GifGridItem.js', () => {
    
    const title = 'Un Titulo';
    const url = 'https://localhost/algo.jp'
    const wrapper = shallow( <GifGridItem title={ title }  url={ url }/> );


    test('debe mostrar el componente correctamente ', () => {
        
        //hacemos una Snapshot
        expect( wrapper ).toMatchSnapshot();

    })

    test('debe de tener un parrafo con el title', () => {
        
        //vemos que si el titulo si muestre lo esperado que es 'title'
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    })

    test('debe de obtener la imagen con los valores pasado por props ', () => {
        
        //const img = wrapper.find('img').at(0) // si hay varias imagen
        const img = wrapper.find('img');
        //console.log( img.props().src );

        //dos forma de hacerlo
        expect( img.props().src ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
        

    })

    test('debe de tener animate__backInLeft ', () => {

        const toSearch = 'animate__backInLeft';
        const div = wrapper.find('div').at(0);
        const className =  div.prop('className')

        //include es para saber si la plabra se encuentra en parrafo
       expect( className.includes( toSearch ) ).toBe( true );

    })
    
})
