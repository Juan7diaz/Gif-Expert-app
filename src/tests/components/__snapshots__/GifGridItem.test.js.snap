// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente GifGridItem.js debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__backInLeft"
>
  <img
    alt="Un Titulo"
    src="https://localhost/algo.jp"
  />
  <div
    className="card-body"
  >
    <p
      className="card-text"
    >
      Un Titulo
    </p>
  </div>
</div>
`;
